### PR TITLE
Test enkf_fs_fwrite_misfit

### LINKED ISSUE #2465: Test and refactor `enkf_fs_fwrite_misfit`
`enkf_fs_fwrite_misfit` is the only function calling `enkf_fs_open_case_file` which is simple and should be removed.
`enkf_fs_open_case_file` calls `util_mkdir_fopen` so this issue is part of the work in replacing `util_mkdir_fopen` with standard C++.

### PR DESCRIPTION
**Pull request by dafeda**
Replaced function call in enkf_fs_fwrite_misfit with code
since this is the only place it's called from.

**Issue**
Resolves #2465

### CODE REVIEW

File: libres/tests/enkf/test_enkf_fs.cpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                                       enkf_obs, fs, ens_size, history_length,
+                                       false);
+
+            THEN("Writing misfits creates a file") {
+                enkf_fs_fwrite_misfit(fs);
+                REQUIRE(std::filesystem::exists(file_path / "files" /
+                                                "misfit-ensemble"));
+            }
+        }
+        // Hack:
```
Comment (dafeda) [inline libres/tests/enkf/test_enkf_fs.cpp:47]: Just how bad of an idea is this @dotfloat ?

File: libres/lib/enkf/enkf_fs.cpp [hunk 2]
```diff
@@ -176,7 +176,7 @@ struct enkf_fs_struct {
 UTIL_SAFE_CAST_FUNCTION(enkf_fs, ENKF_FS_TYPE_ID)
 UTIL_IS_INSTANCE_FUNCTION(enkf_fs, ENKF_FS_TYPE_ID)
 
-static void enkf_fs_umount(enkf_fs_type *fs);
+extern "C" void enkf_fs_umount(enkf_fs_type *fs);
```
Comment (dafeda) [inline libres/lib/enkf/enkf_fs.cpp:179]: Can this be avoided here @dotfloat ?

Comment (dotfloat) [reply libres/lib/enkf/enkf_fs.cpp:179]: Iff. it's not used in Python, which I think it is.

File: libres/tests/enkf/test_enkf_fs.cpp [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+misfit_ensemble_initialize(misfit_ensemble_type *misfit_ensemble,
+                           const ensemble_config_type *ensemble_config,
+                           const enkf_obs_type *enkf_obs, enkf_fs_type *fs,
+                           int ens_size, int history_length, bool force_init);
+void enkf_fs_init_path_fmt(enkf_fs_type *fs);
+void enkf_fs_set_read_only(enkf_fs_type *fs, bool read_only);
+
+TEST_CASE("enkf_fs_fwrite_misfit", "[enkf]") {
+    GIVEN("An instance of enkf_fs") {
+        TmpDir tmpdir;
```
Comment (dotfloat) [inline libres/tests/enkf/test_enkf_fs.cpp:23]: Minor style stuff: I have a macro which does this but doesn't require you to give `TmpDir` a name.
```suggestion
        WITH_TMPDIR;
```

File: libres/tests/enkf/test_enkf_fs.cpp [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        std::filesystem::path file_path = std::filesystem::current_path();
+        enkf_fs_type *fs = enkf_fs_alloc_empty(file_path.c_str());
```
Comment (dotfloat) [inline libres/tests/enkf/test_enkf_fs.cpp:25]: Ignorable, but :pleading_face: 
```suggestion
        auto file_path = std::filesystem::current_path();
        auto fs = enkf_fs_alloc_empty(file_path.c_str());
```

File: libres/tests/enkf/test_enkf_fs.cpp [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        enkf_fs_init_path_fmt(fs);
+
+        WHEN("Misfits ensemble is initialized with minimal config") {
+            misfit_ensemble_type *misfit_ensemble =
+                enkf_fs_get_misfit_ensemble(fs);
+            ensemble_config_type *ensemble_config =
+                ensemble_config_alloc_full("name-not-important");
+            enkf_obs_type *enkf_obs =
+                enkf_obs_alloc(nullptr, nullptr, nullptr, nullptr, nullptr);
```
Comment (dotfloat) [inline libres/tests/enkf/test_enkf_fs.cpp:34]: Likewise
```suggestion
            auto misfit_ensemble = enkf_fs_get_misfit_ensemble(fs);
            auto ensemble_config = ensemble_config_alloc_full("foo");
            auto enkf_obs = enkf_obs_alloc(nullptr, nullptr, nullptr, nullptr, nullptr);
```

Lest we become Java with its
```java
ArrayList arrayList = new ArrayList<>();
```

File: libres/tests/enkf/test_enkf_fs.cpp (hunk 1 above)
Comment (dotfloat) [reply libres/tests/enkf/test_enkf_fs.cpp:47]: I think it's perfectly fine, especially since it's explained in the comment.

**Review by dotfloat — APPROVED** [review]

**Pull request merged**